File: 02_ProgrammingFundamentals/_01_BasicSyntaxConditionalStatementsAndLoops/_01_Lecture/_04_BackIn30Minutes.java
package _02_ProgrammingFundamentals._01_BasicSyntaxConditionalStatementsAndLoops._01_Lecture;

import java.util.Scanner;

public class _04_BackIn30Minutes {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Read input values
        int hours = Integer.parseInt(scan.nextLine());
        int minutes = Integer.parseInt(scan.nextLine());

        // Add variables
        int timePlus30 = hours * 60 + minutes + 30;
        int newHour = timePlus30 / 60;
        int newMinutes = timePlus30 % 60;

        if (newHour > 23) {
            newHour = 0;
        }

        // Print result
        System.out.printf("%d:%02d", newHour, newMinutes);

        scan.close();
    }
}

File: 24_C#Fundamentals/_01_BasicSyntaxConditionalStatementsAndLoops/_01_Lecture/_04_BackIn30Minutes/_04_BackIn30Minutes.cs
using System;

public class _04_BackIn30Minutes
{
    public static void Main(string[] args)
    {
        // Read input values
        int hours = int.Parse(Console.ReadLine());
        int minutes = int.Parse(Console.ReadLine());

        // Add variables
        int timePLus30 = hours * 60 + minutes + 30;
        int newHour = timePLus30 / 60;
        int newMinutes = timePLus30 % 60;

        if (newHour > 23)
        {
            newHour = 0;
        }

        // Print result
        Console.WriteLine($"{newHour}:{newMinutes:D2}");
    }
}
